src/sotauptaneclient.cc

```diff
@@ -238,6 +227,73 @@ void SotaUptaneClient::runForever(command::Channel *commands_channel) {
   }
 }
 
+void SotaUptaneClient::initSecondaries() {
+  std::vector<Uptane::SecondaryConfig>::const_iterator it;
+  for (it = config.uptane.secondary_configs.begin(); it != config.uptane.secondary_configs.end(); ++it) {
+    boost::shared_ptr<Uptane::SecondaryInterface> sec = Uptane::SecondaryFactory::makeSecondary(*it);
+    std::string sec_serial = sec->getSerial();
+    std::map<std::string, boost::shared_ptr<Uptane::SecondaryInterface> >::const_iterator map_it =
+        secondaries.find(sec_serial);
+    if (map_it != secondaries.end()) {
+      LOGGER_LOG(LVL_error, "Multiple secondaries found with the same serial: " << sec_serial);
+      continue;
+    }
+    secondaries[sec_serial] = sec;
+    uptane_repo.addSecondary(sec_serial, sec->getHwId(), sec->getPublicKey());
+  }
+}
+
+struct SerialCompare {
+  std::string target;
+  bool operator()(std::pair<std::string, std::string> &in) { return (in.first == target); }
+};
+
+// Check stored secondaries list against secondaries known to aktualizr via
+// commandline input and legacy interface.
+void SotaUptaneClient::verifySecondaries() {
+  std::vector<std::pair<std::string, std::string> > serials;
+  if (!uptane_repo.storage.loadEcuSerials(&serials) || !serials.size()) {
+    LOGGER_LOG(LVL_error, "No ECU serials found in storage!");
+    return;
+  }
+
+  std::vector<bool> found(serials.size(), false);
+  SerialCompare primary_comp;
+
+  primary_comp.target = uptane_repo.getPrimaryEcuSerial();
+  // Should be a const_iterator but we need C++11 for cbegin.
+  std::vector<std::pair<std::string, std::string> >::iterator store_it;
+  store_it = std::find_if(serials.begin(), serials.end(), primary_comp);
```

Comment:
Why not just append to the list ecus that have not been found? It could be more compact end efficient.
Also we already prints ecus that was not found but at the and of function we iterates again and displays ecus that was not found again

Reply:
Append the missing serials to which list? And where else do we print ECUs that weren't found?

There are two things we are checking for here. First, that we aren't seeing any new serials that weren't seen before. (We don't currently support that.) Second, that all of the serials previously stored are still present. (That is more or less fine but we should alert the user.) I print an error for both cases.